docs/concepts/filesystems.md

```diff
@@ -89,45 +89,76 @@ minio_block = RemoteFileSystem(
 minio_block.save("minio")
 ```
 
+## Azure
 
-## S3
-
-The `S3` file system block enables interaction with Amazon S3. Under the hood, `S3` uses [`s3fs`](https://s3fs.readthedocs.io/en/latest/).
+The `Azure` file system block enables interaction with Azure Datalake and Azure Blob Storage. Under the hood, `Azure` uses [`adlfs`](https://github.com/fsspec/adlfs).
 
-`S3` properties include:
+`Azure` properties include:
 
 | Property | Description |
 | --- | --- |
 | basepath | String path to the location of files on the remote filesystem. Access to files outside of the base path will not be allowed. |
-| aws_access_key_id | AWS Access Key ID |
-| aws_secret_access_key | AWS Secret Access Key |
+| azure_storage_connection_string | Azure storage connection string. |
+| azure_storage_account_name | Azure storage account name. |
+| azure_storage_account_key | Azure storage account key. |
 
 
 To create a block:
 
 ```python
-from prefect.filesystems import S3
+from prefect.filesystems import Azure
 
-block = S3(basepath="my-bucket/folder/")
+block = Azure(basepath="my-bucket/folder/")
 block.save("dev")
 ```
 
 To use it in a deployment:
+
+<div class="terminal">
 ```bash
-prefect deployment build path/to/flow.py:flow_name --name deployment_name --tag dev -sb s3/dev
+prefect deployment build path/to/flow.py:flow_name --name deployment_name --tag dev -sb az/dev
 ```
+</div>
 
-You need to install `s3fs`to use it.
+You need to install `adlfs` to use it.
+
+## GitHub
+
+The `GitHub` filesystem block enables interaction with GitHub repositories. 
```

Comment:
```suggestion
The `GitHub` filesystem block enables interaction with GitHub repositories. This block is currently read-only and limited to public repositories.
```

Reply:
I know there's an open issue for private repos and a hack, so might want to exclude the public repo section I added.